vendor/bat-native-ads/src/bat/ads/notification_ad_info.cc

```diff
@@ -33,98 +35,91 @@ bool NotificationAdInfo::IsValid() const {
+    creative_instance_id = *value;
   }
 
-  if (document.HasMember("creative_instance_id")) {
-    creative_instance_id = document["creative_instance_id"].GetString();
+  if (const auto* value = root.FindString("creative_set_id")) {
+    creative_set_id = *value;
   }
 
-  if (document.HasMember("creative_set_id")) {
-    creative_set_id = document["creative_set_id"].GetString();
+  if (const auto* value = root.FindString("campaign_id")) {
+    campaign_id = *value;
   }
 
-  if (document.HasMember("campaign_id")) {
-    campaign_id = document["campaign_id"].GetString();
+  if (const auto* value = root.FindString("advertiser_id")) {
+    advertiser_id = *value;
   }
 
-  if (document.HasMember("advertiser_id")) {
-    advertiser_id = document["advertiser_id"].GetString();
+  if (const auto* value = root.FindString("segment")) {
+    segment = *value;
   }
 
-  if (document.HasMember("segment")) {
-    segment = document["segment"].GetString();
+  if (const auto* value = root.FindString("title")) {
+    title = *value;
   }
 
-  if (document.HasMember("title")) {
-    title = document["title"].GetString();
+  if (const auto* value = root.FindString("body")) {
+    body = *value;
   }
 
-  if (document.HasMember("body")) {
-    body = document["body"].GetString();
-  }
-
-  if (document.HasMember("target_url")) {
-    target_url = GURL(document["target_url"].GetString());
+  if (const auto* value = root.FindString("target_url")) {
+    target_url = GURL(*value);
   }
 
   return true;
 }
 
-void SaveToJson(JsonWriter* writer, const NotificationAdInfo& info) {
-  writer->StartObject();
-
-  writer->String("type");
-  writer->String(info.type.ToString().c_str());
-
-  writer->String("uuid");
-  writer->String(info.placement_id.c_str());
-
-  writer->String("creative_instance_id");
-  writer->String(info.creative_instance_id.c_str());
-
-  writer->String("creative_set_id");
-  writer->String(info.creative_set_id.c_str());
-
-  writer->String("campaign_id");
-  writer->String(info.campaign_id.c_str());
-
-  writer->String("advertiser_id");
-  writer->String(info.advertiser_id.c_str());
-
-  writer->String("segment");
-  writer->String(info.segment.c_str());
-
-  writer->String("title");
-  writer->String(info.title.c_str());
+std::string NotificationAdInfo::ToJson() const {
+  std::string json;
+  base::JSONWriter::Write(ToValue(), &json);
+  return json;
+}
 
-  writer->String("body");
-  writer->String(info.body.c_str());
+bool NotificationAdInfo::FromJson(const std::string& json) {
+  auto document = base::JSONReader::ReadAndReturnValueWithError(
+      json, base::JSON_PARSE_CHROMIUM_EXTENSIONS |
+                base::JSONParserOptions::JSON_PARSE_RFC);
+
+  if (!document.value.has_value()) {
+    BLOG(1, "Invalid notification ad info. json="
+                << json << ", error line=" << document.error_line
+                << ", error column=" << document.error_column
+                << ", error message=" << document.error_message);
```

Comment:
can we use `NOTREACHED()` instead of logging here as this should never been reached. Also see other BLOG

Reply:
Fixed